tests/Unit/Models/CourseTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Tests\Unit\Models;
+
+use Tests\TestCase;
+use App\Models\Course;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class CourseTest extends TestCase
+{
+    protected  $course;
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+        $this->course = new Course();
+    }
+
+    protected function tearDown() : void
+    {
+        parent::tearDown();
+        $this->course = null;
+    }
```

Comment:
2 function này dùng để làm gì? nó chạy ntn?

Reply:
Hàm setUp dùng để khởi tạo các biến số cho toàn bộ class test
Hàm tearDown dùng để unset các biến , ngược lại với hàm setUp
Chu trình chạy setUp()-> test_function_a()->tearDown()->setUp()->test_function_b()->tearDown()->...